tools/docker/jetson/get_debs.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import subprocess
+import sys
+
+import apt
+
+if len(sys.argv) < 2:
+    print("Please specify a package to install", file=sys.stderr)
+    sys.exit(1)
+
+packages = str(subprocess.check_output(["apt-rdepends"] + sys.argv[1:]))
+
+packages = [line for line in packages.split("\n") if "Depends:" not in line]
+
+
+cache = apt.Cache()
+
+final = []
+for package in packages:
+    if package.strip() == '':
+        continue
+    if cache.is_virtual_package(package):
+        providers = cache.get_providing_packages(package)
+        if len(providers) == 1:
+            # print("Virtual: " + package)
+            provider = providers[0]
+            if(provider.architecture() not in ['arm64', 'all']):
+                if cache.has_key(provider.name + ':arm64'):
+                    final.append(cache[provider.name + ':arm64'])
+                elif cache.has_key(provider.name + ':all'):
+                    final.append(cache[provider.name + ':all'])
+                else:
+                    print(package + " not found for arm64 architecture. Ignoring", file=sys.stderr)
+        else:
+            arm_providers = [provider for provider in providers
+                             if provider.architecture() == 'arm64']
+            if len(arm_providers) == 1:
+                final.append(arm_providers[0])
+            elif len(arm_providers) > 1:
```

Comment:
Is this the best way of doing it?
If it is the best way then Isn't the else if and else the same case?

Reply:
No, the else case is the length 0 case